frontend/components/PDPSheet/index.spec.jsx

```diff
@@ -0,0 +1,151 @@
+    mockedRouteDidChangeSubscribe.mockReset();
+  });
+
+  it('should do nothing when productId is missing', () => {
+    mockedCurrentBaseProductId = null;
+    mockedConfig.productPageAddToCart = {
+      type: 'mockedType',
+      headline: 'mockedHeadline',
+    };
+
+    const component = makeComponent();
+    const instance = component.find('PDPSheet').instance();
+
+    expect(component.html()).toBe(null);
+    expect(instance.state.disabled).toBe(false);
+    expect(instance.state.isOpen).toBe(false);
+    expect(mockedPDPAddToCartSuccessSubscribe).toHaveBeenCalled();
+    expect(mockedRouteDidChangeSubscribe).toHaveBeenCalled();
+
+    // Try to open without current productId
+    mockedPDPAddToCartSuccessSubscribe.mock.calls[0][0]();
+    // Should be closed.
+    expect(instance.state.isOpen).toBe(false);
+
+    // Should not fetch on update
+    instance.componentDidUpdate({ productId: null });
+    expect(mockedGetProductRelations).not.toHaveBeenCalled();
+
+    // Component not update
+    expect(instance.shouldComponentUpdate({ productId: null }, { isOpen: false })).toBe(false);
+  });
+
+  it('should do nothing when disabled', () => {
+    mockedCurrentBaseProductId = 'mockedId';
+    mockedConfig.productPageAddToCart.type = null;
+    const component = makeComponent();
+    const instance = component.find('PDPSheet').instance();
+
+    expect(component.html()).toBe(null);
+    expect(instance.state.disabled).toBe(true);
+    expect(instance.state.isOpen).toBe(false);
+    expect(mockedPDPAddToCartSuccessSubscribe).not.toHaveBeenCalled();
+    expect(mockedRouteDidChangeSubscribe).not.toHaveBeenCalled();
+
+    // Should not fetch on update
+    instance.componentDidUpdate({ productId: null });
+    expect(mockedGetProductRelations).not.toHaveBeenCalled();
+
+    // Component should never update
+    expect(instance.shouldComponentUpdate()).toBe(false);
+  });
+
+  describe('Life cycle', () => {
+    let component;
+    let instance;
+    let routeDidChangeFunc;
+    let addToCartSuccessFunc;
+
+    it('should render Sheet and update when product changes', () => {
+      mockedCurrentBaseProductId = 'mockedId';
+      mockedConfig.productPageAddToCart.type = 'mockedType';
+      component = makeComponent();
+      instance = component.find('PDPSheet').instance();
+
+      expect(component.find('MockedSheet').exists()).toBe(true);
+      expect(instance.state.disabled).toBe(false);
+      expect(instance.state.isOpen).toBe(false);
+      expect(mockedGetProductRelations).toHaveBeenCalled();
+
+      // eslint-disable-next-line prefer-destructuring
+      routeDidChangeFunc = mockedRouteDidChangeSubscribe.mock.calls[0][0];
+      // eslint-disable-next-line prefer-destructuring
+      addToCartSuccessFunc = mockedPDPAddToCartSuccessSubscribe.mock.calls[0][0];
+    });
+
+    it('should react on add to cart', () => {
+      // Simulate add to cart event
+      addToCartSuccessFunc();
+      expect(instance.state.isOpen).toBe(true);
```

Comment:
you should add a check here to test if the component really renders the correct content

Reply:
done